### PR TITLE
Noam/bugfix/token class bugfixes

### PR DESCRIPTION
**Pull request by noamzbr**
Related to https://github.com/deepchecks/deepchecks/issues/2529 but fixes only part of it

### CODE REVIEW

File: deepchecks/nlp/utils/nlp_plot.py [hunk 1]
```diff
@@ -259,12 +280,27 @@ def two_datasets_scatter_plot(plot_title: str, plot_data: pd.DataFrame, train_da
...
-        plot_data['Label'] = list(train_dataset.label_for_display(model_classes=model_classes)) + \
-                             list(test_dataset.label_for_display(model_classes=model_classes))
+
+    if train_dataset.task_type == TaskType.TOKEN_CLASSIFICATION:
+        plot_data['Sample'] = np.concatenate([train_dataset.tokenized_text, test_dataset.tokenized_text])
+
+        if train_dataset.has_label():
+            plot_data['Label'] = list(train_dataset.label_for_display(model_classes=model_classes)) + \
+                                 list(test_dataset.label_for_display(model_classes=model_classes))
+            plot_data['Sample'] = annotated_token_classification_text(plot_data['Sample'], plot_data['Label'])
```
Comment (nirhutnik) [inline deepchecks/nlp/utils/nlp_plot.py:290]: I'm confused, why show the label together with the text when you're showing the sample for token classification?
And if you add that... Why show the counts for the labels?

File: deepchecks/utils/numbers.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""General utilities for working with numbers."""
+
+import numpy as np
+
+
+def round_sig(x: float, sig: int = 2):
+    """Round a number to a given number of significant figures."""
```
Comment (nirhutnik) [inline deepchecks/utils/numbers.py:17]:
```suggestion
    """Round a number to a given number of significant digits."""
```
I suppose you meant that?

**Review by nirhutnik — APPROVED** [review]
